personio-client/src/Personio/Types.hs

```
@@ -641,6 +697,86 @@ validatePersonioEmployee = withObjectDump "Personio.Employee" $ \obj -> do
                 Nothing -> tell [FlowdockInvalid]
                 Just _  -> pure ()
 
+        emailValidate :: WriterT [ValidationMessage] Parser ()
+        emailValidate = do
+            email <- lift (parseAttribute obj "email")
+            case match emailRegexp email of
+                Nothing -> tell [EmailInvalid email]
+                Just _  -> pure ()
+        
+        supervisorValidate :: WriterT [ValidationMessage] Parser ()
+        supervisorValidate = do
+            position <- lift (parseAttribute obj "position")
+            case match noSuperRegexp position of
+                Nothing -> attributeObjectMissing "supervisor" SupervisorMissing
+                Just _  -> pure ()
+          where
+            noSuperRegexp = string "CEO"
+
+        hoursValidate :: WriterT [ValidationMessage] Parser ()
+        hoursValidate = do
+            hours <- lift (parseAttribute obj "weekly_working_hours")
+            case toNum hours of
+                Just n  -> if n > 0
+                    then pure ()
+                    else tell [HoursInvalid n]
+                Nothing -> lift (typeMismatch "weekly_working_hours" hours)
+          where
+            toNum :: Value -> Maybe Scientific
+            toNum (String v) = readMaybe $ T.unpack v
```

I hope it won't ever, so let it just fail in that case.